reactive-streams/src/test/scala/fs2/interop/reactivestreams/CancellationSpec.scala

```diff
@@ -36,54 +35,44 @@ import java.util.concurrent.atomic.AtomicBoolean
   * failures due to race conditions more repeatable
   */
 class CancellationSpec extends Fs2Suite {
-
-  case class Sub[A](b: AtomicBoolean) extends Subscriber[A] {
+  final class Sub[A](b: AtomicBoolean) extends Subscriber[A] {
     def onNext(t: A) = b.set(true)
     def onComplete() = b.set(true)
     def onError(e: Throwable) = b.set(true)
-    def onSubscribe(s: Subscription) = b.set(true)
+    def onSubscribe(s: Subscription) = ()
```

Comment:
Why this change?

Reply:
`onSubscribe` must always be called, so if we set the _flag_ there the test would always fail.
Previously it worked because it wasn't even subscribing.